File: src/main/java/icu/takeneko/omms/central/network/session/response/ControllerData.java
package icu.takeneko.omms.central.network.session.response;

import icu.takeneko.omms.central.controller.Controller;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

@Getter
public class ControllerData {
    //{\"name\":\"creative\",\"executable\":\"python\",\"type\":\"fabric\",\"launchParams\":\"-m mcdreforged\",\"workingDir\":\"..\\\\creative\",\"httpQueryAddress\":\"127.0.0.1:50014\",\"statusQueryable\":true}
    private ControllerData() {
    }

    private String name;
    private String type;
    private String displayName;
    private boolean statusQueryable;

    public static @NotNull ControllerData fromController(@NotNull Controller controller) {
        var data = new ControllerData();
        data.name = controller.getName();
        data.statusQueryable = controller.isStatusQueryable();
        data.displayName = controller.getDisplayName();
        data.type = controller.getType();
        return data;
    }

}

File: src/main/java/icu/takeneko/omms/central/controller/ControllerData.kt
package icu.takeneko.omms.central.controller

@kotlinx.serialization.Serializable
data class ControllerData(
    val name: String,
    val displayName: String = name,
    val type: String,
    val httpQueryAddress: String,
    val statusQueryable: Boolean
){
    fun inflate():Controller{
        return ControllerImpl(
            name,
            displayName,
            type,
            httpQueryAddress,
            statusQueryable
        )
    }
}
